internal/services/dispatcher/server_v1.go

```diff
@@ -1318,11 +1330,37 @@ func (s *DispatcherImpl) isMatchingWorkflowRunV1(msg *msgqueue.Message, acks *wo
 		}
 
 		if len(res) == 0 {
-			return nil, false
+			return nil, nil
+		}
+
+		return res, nil
+	case "task-completed":
+		res := make([]*contracts.WorkflowRunEvent, 0)
+
+		payloads := msgqueue.JSONConvert[tasktypes.CompletedTaskPayload](msg.Payloads)
+
+		for _, payload := range payloads {
+			if payload.IsDAG && acks.hasWorkflowRun(payload.WorkflowRunId) {
+				output := string(payload.Output)
+
+				res = append(res, &contracts.WorkflowRunEvent{
+					WorkflowRunId:  payload.WorkflowRunId,
+					EventType:      contracts.WorkflowRunEventType_WORKFLOW_RUN_EVENT_TYPE_FINISHED,
+					EventTimestamp: timestamppb.New(time.Now()),
+					Results: []*contracts.StepRunResult{
+						{
+							StepRunId:      payload.ExternalId,
+							StepReadableId: payload.StepReadableId,
+							JobRunId:       payload.ExternalId,
+							Output:         &output,
+						},
+					},
+				})
```

Comment:
This is the core change in this file, if we have a single-task workflow and we get a task-completed event, we send it immediately to the worker. 

I think there's a chance of race conditions here, because we don't want to send a message to a worker before its been fully written/acknowledged by the database, which we risk here. 

So if we were to fail to write the `task-completed` event to the database, we won't successfully release the task, and it may have a different status after we hit a timeout/reassignment than the parent workflow would see. 

Yet another case where 2PC is necessary - or perhaps just moving the processing of the task-completed event to the gRPC layer like we used to have. I need to think a bit more about it.